### PR TITLE
feat(app): show offset deletion warning if previous offsets exist

### LINKED ISSUE #8841: LPC Deletion Warning Follow-ons
Nice to have follow up: only show the alert "Once you begin Labware Position Check, previously created Labware Offsets will be discarded." if previous offsets exist

### PR DESCRIPTION
**Pull request by smb2268**
close #8841

# Overview

Show alert warning on LPC intro screen indicating previous offsets will be deleted _only_ if previous offsets exist
# Changelog
Put alert behind check for existing labware offsets

# Review requests

Confirm alert is present when previous offsets exist
Confirm alert is not present when no offsets exist
# Risk assessment

Low

### CODE REVIEW

**Review by jerader — COMMENTED** [review]
I was able to get through the flow a few times and the modal works as expected when 1) offsets are added and 2) offsets are not added. However, if you go through the flow once and change 1 thing, say X = 0.1, then if you go through the flow again and make X = 0, all the offsets are technically 0 but the banner will still show up if you do LPC again. Not sure if this is the intended behavior or not?

**Comment by smb2268** [comment]
> I was able to get through the flow a few times and the modal works as expected when 1) offsets are added and 2) offsets are not added. However, if you go through the flow once and change 1 thing, say X = 0.1, then if you go through the flow again and make X = 0, all the offsets are technically 0 but the banner will still show up if you do LPC again. Not sure if this is the intended behavior or not?

This is because we just check whether the current run has offsets on it, so even if you explicitly set the offset to 0,  an offset will technically exist. @shlokamin do you think I should modify the `getLatestLabwareOffsetCount` function to only count offsets if they aren't 0?

**Comment by smb2268** [comment]
> I was able to get through the flow a few times and the modal works as expected when 1) offsets are added and 2) offsets are not added. However, if you go through the flow once and change 1 thing, say X = 0.1, then if you go through the flow again and make X = 0, all the offsets are technically 0 but the banner will still show up if you do LPC again. Not sure if this is the intended behavior or not?

This should be fixed with the most recent commit! cc @shlokamin @jerader

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts [hunk 1]
```diff
@@ -11,25 +11,35 @@ const TRASH_LOCATION = { slotName: '12' }
 export const getLatestLabwareOffsetCount = (
   labwareOffsets: LabwareOffset[]
 ): number => {
-  const uniqueLabwareInstances = labwareOffsets.reduce<Set<string>>(
-    (uniqueLabwareInstances, offset) => {
+  // labware offsets are in order of creation, we want to use the most recent ones
+  const orderedLabwareOffsets = labwareOffsets
+    .slice()
+    .sort((a, b) => a.date - b.date)
```
Comment (shlokamin) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts]: `createdAt` is what's coming from the run record, not `date`:

```js
[
  {
    "id": "2fe7c0fa-72c5-4d7f-8724-4cc96ec0608b",
    "createdAt": "2021-11-29T23:24:07.524221+00:00",
    "definitionUri": "opentrons/opentrons_1_trash_1100ml_fixed/1",
    "location": {
      "slotName": "12"
    },
    "vector": {
      "x": 0,
      "y": 0,
      "z": 0
    }
  },
]
```

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import isEqual from 'lodash/isEqual'
```
Comment (shlokamin) [inline app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts:1]: looks like we need to update the type of `LabwareOffset` to include `createdAt`

File: app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts (hunk 1 above)
Comment (smb2268) [reply app/src/organisms/ProtocolSetup/LabwarePositionCheck/utils/getLatestLabwareOffsetCount.ts]: 🤦🏼‍♀️

**Review by jerader — APPROVED** [review]
tested on my robot, lgtm 🌴

**Pull request merged**